src/github.com/travis-ci/jupiter-brain/vsphere.go

```diff
@@ -264,6 +280,16 @@ func (i *vSphereInstanceManager) createClient(ctx context.Context) (*govmomi.Cli
 	}
 
 	client.Client.RoundTripper = vim25.Retry(client.Client.RoundTripper, vim25.TemporaryNetworkError(3))
+	client.Client.RoundTripper = &soapBreakerRoundTripper{
+		rt: client.Client.RoundTripper,
+		cb: gobreaker.NewCircuitBreaker(gobreaker.Settings{
+			Name: "vSphere govmomi",
+			ReadyToTrip: func(counts gobreaker.Counts) bool {
+				failureRatio := float64(counts.TotalFailures) / float64(counts.Requests)
+				return counts.Requests >= 3 && failureRatio >= 0.6
```

Comment:
For example, right now, if we have Jupiter Brain chugging along and it's made 1,000 passing requests, and then every request starts erroring, the breaker won't trip until we've had 1,500 requests that error.